widgets/heatmap/heatmap.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package heatmap contains a widget that displays heat maps.
+package heatmap
+
+import (
+	"errors"
+	"image"
+	"sync"
+
+	"github.com/mum4k/termdash/cell"
+	"github.com/mum4k/termdash/private/canvas"
+	"github.com/mum4k/termdash/terminal/terminalapi"
+	"github.com/mum4k/termdash/widgetapi"
+	"github.com/mum4k/termdash/widgets/heatmap/internal/axes"
+)
+
+// HeatMap draws heat map charts.
+//
+// Heatmap consists of several cells. Each cell represents a value.
+// The larger the value, the darker the color of the cell (from white to black).
```

Comment:
Actually, I have tried red color, but I found that the effect was not ideal. The color difference between different values was not obvious. This is because the number of red in Xterm colors is limited (only about a dozen from light red to dark red). However there are more colors from white to gray to black colors (more than 20 kinds), which can better reflect the difference between the values.

So if you want to make the colors configurable, it requires the terminal supports 'true color'.

Reply:
That is a good point, I understand now why you chose to use the shades of grey and it does make sense.